### ISSUE TITLE
READ-CHAR error on UTF-32BE stream caus

### ISSUE THREAD

**Issue by robert-dodier** [Clozure/ccl#115]
`(read-char s)` where `s` is a UTF-32BE stream causes an error: `Error: #'CCL::%PRIVATE-IOBLOCK-READ-SWAPPED-U16-ENCODED-CHAR is not of type (OR SYMBOL FUNCTION), and can't be FUNCALLed or APPLYed`

UTF-8, UTF-16BE, UTF-16LE, and UTF-32LE are all OK, only UTF-32BE causes an error.

Here's a transcript. Note that `rectangular.csv-utf-32be` is a file with UTF-32BE-encoded text.

```
$ lx86cl
Clozure Common Lisp Version 1.11.5/v1.11.5  (LinuxX8632)
For more information about CCL, please see http://ccl.clozure.com.
CCL is free software.  It is distributed under the terms of the Apache
Licence, Version 2.0.
? (defvar s (open "rectangular.csv-utf-32be" :external-format :utf-32be))
S
? s
#<BASIC-FILE-CHARACTER-INPUT-STREAM ("rectangular.csv-utf-32be"/4 UTF-32BE) #x18249126>
? (eq (peek-char nil s nil) #\U+FEFF)
T
? (read-char s)
> Error: #'CCL::%PRIVATE-IOBLOCK-READ-SWAPPED-U16-ENCODED-CHAR is not of type (OR SYMBOL FUNCTION), and can't be FUNCALLed or APPLYed
> While executing: READ-CHAR, in process listener(1).
> Type :POP to abort, :R for a list of available restarts.
> Type :? for other options.
1 > 
```

`uname -a` reports: `Linux freekbox 3.13.0-44-generic #73-Ubuntu SMP Tue Dec 16 00:23:46 UTC 2014 i686 i686 i686 GNU/Linux`

Clozure CCL info: `lx86cl --version` reports `Version 1.11.5/v1.11.5  (LinuxX8632)`.

```
? *features*
(:PRIMARY-CLASSES :COMMON-LISP :OPENMCL :CCL :CCL-1.2 :CCL-1.3 :CCL-1.4 :CCL-1.5 :CCL-1.6 :CCL-1.7 :CCL-1.8 :CCL-1.9 :CCL-1.10 :CCL-1.11 :CLOZURE :CLOZURE-COMMON-LISP :ANSI-CL :UNIX :OPENMCL-UNICODE-STRINGS :IPV6 :OPENMCL-NATIVE-THREADS :OPENMCL-PARTIAL-MOP :MCL-COMMON-MOP-SUBSET :OPENMCL-MOP-2 :OPENMCL-PRIVATE-HASH-TABLES :X8632-TARGET :X8632-HOST :X86 :X86-TARGET :X86-HOST :LINUX-HOST :LINUX-TARGET :LINUXX86-TARGET :LINUXX8632-TARGET :LINUXX8632-HOST :32-BIT-TARGET :32-BIT-HOST :LINUX :LITTLE-ENDIAN-TARGET :LITTLE-ENDIAN-HOST)
```

**Comment by xrme** [comment]
This is probably due to https://github.com/Clozure/ccl/blob/master/level-1/l1-streams.lisp#L2799

It looks like there is more than one thing wrong on that line.  First, there's an extra `#\'` in front, and second, it looks like the function in question should be `%private-ioblock-read-swapped-u32-encoded-char`.

**Comment by xrme** [comment]
This should be fixed now.  I cherry-picked the fix to the 1.11 branch, too.

Please update (via git pull) and rebuild the lisp with `(rebuild-ccl :clean t)` to pick up the fix.

Thank you for the bug report.
